src/dwt.rs

```diff
@@ -0,0 +1,219 @@
+//! Debug and trace and stuff
+
+use crate::rcc::Clocks;
+use crate::time::Hertz;
+use cortex_m::peripheral::{DCB, DWT};
+use embedded_hal::blocking::delay::{DelayMs, DelayUs};
+
+pub trait DwtExt {
+    fn constrain(self, dcb: DCB, clocks: Clocks) -> Dwt;
+}
+impl DwtExt for DWT {
+    /// Enable trace unit and cycle counter
+    fn constrain(mut self, mut dcb: DCB, clocks: Clocks) -> Dwt {
+        cortex_m::interrupt::free(|_| {
```

Comment:
Why interrupt free if you have ownership of the registers?

Reply:
ups.. I removed this now. It was just some copy-pasta (especially the enable fmc-clock on the next line)